index.php

```diff
@@ -250,138 +250,215 @@
     }
-    //If one column has multiple sorts, combine them into one entry for that column.
+    // If one column has multiple sorts, combine them into one entry for that column.
     $hcolumns[$column] = implode('/', $final);
 }
 
 $table = new html_table();
 $table->head = $hcolumns;
 $table->attributes["class"] = "admintable generaltable";
 $table->data = [];
-foreach($records as $record) {
+foreach ($records as $record) {
     $final = new stdClass();
-    if (IS_ADMIN) {
+    if (REPORT_COURSECOMPLETION_IS_ADMIN) {
         $finalname = fullname($record);
-        $final->fullname = add_link('/user/view.php', array('id' =>$record->userid), $finalname);
+        $final->fullname = add_link('/user/view.php', array('id' => $record->userid), $finalname);
         $final->email = $record->email;
     }
-    $final->course = add_link('/course/view.php', array('id' =>$record->course), $record->fullname);
+    $final->course = add_link('/course/view.php', array('id' => $record->course), $record->fullname);
     $final->timestarted = time_format($record->timestarted);
     $final->timecompleted = time_format($record->timecompleted);
     $final->completionstatus = $record->timecompleted ? 'Yes' : 'No';
     $table->data[] = $final;
 }
 
 echo html_writer::table($table);
-$buttonurl = new moodle_url("index.php", array("sort" => $sort, "dir" => $dir, "perpage" => $perpage, "page" => $page, "export"=>1));
+$buttonurl = new moodle_url("index.php", array(
+    "sort" => $sort,
+    "dir" => $dir,
+    "perpage" => $perpage,
+    "page" => $page,
+    "export" => 1
+    )
+);
 $buttonstring = get_string('exportbutton', 'report_coursecompletion');
 echo $OUTPUT->single_button($buttonurl, $buttonstring);
 echo $OUTPUT->footer();
 
-function add_condition_connectors(&$data, &$where, &$params, $force_and = false) {
-    if(!empty($params)) {
-        if($force_and || (isset($data->operator) && $data->operator == 0)) {
+/**
+ * Appends AND / OR operators to data fields that are assigned to $where param.
+ * @param object $data Form data stored in session.
+ * @param string $where Where clause.
+ * @param array $params Typically stores user ID and cohort ID.
+ * @param string $forceand A nonempty value would convert an OR operator
+ * to AND.
+ */
+function add_condition_connectors(
+    &$data, &$where, &$params, $forceand) {
+    $forceand = set_false_if_empty($forceand);
+
+    if (!empty($params)) {
+        if ($forceand || (isset($data->operator) && $data->operator == 0)) {
             $where .= " AND ";
         } else {
             $where .= " OR ";
         }
     }
 }
 
+/**
+ * Generates a simple html link after constructing a new instance of moodle_url.
+ * @param string $url The URL.
+ * @param string $params These params override current params in moodle_url().
+ * or add new ones.
+ * @param string $string The contents that go between opening and closing tags.
+ * @return string HTML fragment.
+ */
 function add_link($url, $params, $string) {
```

Comment:
this function seems a bit pointless - might be worth just deleting it and calling the html_writer:: directly in the code where it's used.